packages/FormElement/src/FormElement.js

```diff
@@ -1,93 +1,18 @@
 import React from "react";
 import PropTypes from "prop-types";
 import * as constants from "@paprika/constants/lib/Constants";
-import extractChildren from "@paprika/helpers/lib/extractChildren";
-import isNil from "lodash/isNil";
-import uuidv4 from "uuid/v4";
-
-import Description from "./components/Description";
-import Instructions from "./components/Instructions";
-import Content from "./components/Content";
-import ErrorMessage from "./components/ErrorMessage";
-import Help from "./components/Help";
-import Label from "./components/Label";
 import * as sc from "./FormElement.styles";
 
-const subComponentDisplayNames = {
-  Error: "FormElement.Error",
-  Description: "FormElement.Description",
-  Content: "FormElement.Content",
-  Instructions: "FormElement.Instructions",
-  Help: "FormElement.Help",
-};
+export const FormElementContext = React.createContext({});
 
 function FormElement(props) {
-  const {
-    children,
-    hasOptionalLabel,
-    hasRequiredLabel,
-    id,
-    isDisabled,
-    isInline,
-    isLabelVisuallyHidden,
-    label,
-    size,
-    hasFieldSet,
-    onClickLabel,
-    ...moreProps
-  } = props;
-
-  const extractedChildren = extractChildren(children, Object.values(subComponentDisplayNames));
-  const ariaDescriptionId = React.useRef(uuidv4()).current;
-  const ariaErrorId = React.useRef(uuidv4()).current;
-  const ariaInstructionsId = React.useRef(uuidv4()).current;
-  const uniqueInputId = React.useRef(uuidv4()).current;
-  const hasError =
-    !!extractedChildren[subComponentDisplayNames.Error] &&
-    !!extractedChildren[subComponentDisplayNames.Error].props.children;
-
-  const generateLabelId = id => (isNil(id) || id === "" ? uniqueInputId : id);
-  const idForLabel = generateLabelId(id);
-  const refLabel = React.useRef(null);
+  const { children, isDisabled, isInline, size, hasFieldSet, formElementA11yProps, ...moreProps } = props;
 
-  const getClonedElement = (displayName, extraProps = {}) => {
-    if (!extractedChildren[displayName]) return null;
-    return React.cloneElement(extractedChildren[displayName], extraProps);
+  const value = {
+    formElementA11yProps,
+    hasFieldSet,
```

Comment:
Wouldn't it make sense to put the rest of the props – `isInline`, `size`, `isDisabled` on context too for any of the sub components that may rely on those for styling adjustments?

Reply:
Currently I don't see any other sub components making use of those props so don't need to add them to context right now